[DOC] Describe how to propose a migration script to the community. (#4227)
OCA/OpenUpgrade: docsource/090_contribute.rst

  database, and give back any code improvements developed during the project.

How to contribute new migration scripts
---------------------------------------

To Contribute to OpenUpgrade you must make one pull request per module.

For example, if you want to propose the migration script of the ``account``
module from version 15.0 to version 16.0:

  * Always ensure that there is no work already in progress by a community member.

>     For that purpose, go the issue named "Migration to version 16.0" on github

Reviewer: ```suggestion
    For that purpose, go the issue named "Migration to version NN.0" on GitHub, where NN is the version for which you want to propose the script.
```

>     `here <https://github.com/OCA/OpenUpgrade/issues/3681/>`_.

Reviewer: ```suggestion
    `here <https://github.com/OCA/OpenUpgrade/issues/3681/>`_.
    
    Sometimes, the pull request is not yet registered in the issue, so it's not bad to do a search on the opened pull requests with the name of the module to look for.
```

>   * Then, ensure that all the dependencies of the module are mark as ``done`` or

Reviewer: This is not a requirement for developing the script. Just to merge it.

    ``Nothing to do`` in OpenUpgrade for that version.
    For that purpose, refer to :doc:`030_coverage_analysis`.
    In our example, check the ``depends`` key of the `manifest <https://github.com/odoo/odoo/blob/16.0/addons/account/__manifest__.py#L18>`_ of the ``account`` module 
    If some dependencies are missing, you should start by migrating these modules.

  * As other OCA contribution, create a new branch, from an up to date OCA branch:

    ``git checkout -b 16.0-mig-account``

>   * Make a copy of the analysis file.

Reviewer: ```suggestion
  * Make a copy of the analysis file with the suffix ``_work``.
```

    ``cd ./openupgrade_scripts/scripts/account/16.0.1.2/``
    ``cp upgrade_analysis.txt upgrade_analysis_work.txt``

>   * Explain the changes to do in the ``upgrade_analysis_work.txt``.

Reviewer: Explain here about the format of the comments:

- `NOTHING TO DO`, and why it's nothing to do
- `DONE: pre-migration/post-migration:` indicating the what and maybe the why.
- `TODO:` This shouldn't be usually done.

Then, also explain that they can reorder lines and group them together for including all of them in a "logical" block for putting only one comment including all of them. Examples: all the noupdate=0 new and del ir* records, or the `NEW` and `DEL` lines for a field that is renamed.

>     Write ``pre-migration.py`` and / or ``post-migration.py`` scripts in the same folder.

Reviewer: Put this as new points instead of the same one.

    Comment / uncomment lines in ``noupdate_changes.xml``.
    (Read more in :doc:`080_migration_script_development`)

  * Finally, commit your changes.
    ``git add . && git commit -am "[OU-ADD] account"``

>     (For a fix, use ``[OU-FIX]`` prefix)

Reviewer: Put the possible tags in a separate line explaining it: `[OU-ADD]`, `[OU-FIX]`, `[OU-IMP]`. We should propose these tags also to https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#71commit-message

  * Propose your changes to the community for review.

>     ``git push MY_REMOTE 16.0-mig-account``

Reviewer: and open the pull request.

  database, and give back any code improvements developed during the project.

How to contribute new migration scripts
---------------------------------------

To Contribute to OpenUpgrade you must make one pull request per module.

For example, if you want to propose the migration script of the ``account``
module from version 15.0 to version 16.0:

  * Always ensure that there is no work already in progress by a community member.
    For that purpose, go the issue named "Migration to version NN.0" on GitHub, where NN is the version for which you want to propose the script.

>     `here <https://github.com/OCA/OpenUpgrade/issues/3681/>`_.

Reviewer: Are we going to keep this issue reference? It won't go well through time, when we were on Odoo 20 (or just already in 17).

Author: Well, the sentence before is "For example, if you want to propose the migration script of the ``account``
module from version 15.0 to version 16.0" so it make sense.
but well, replaced by "https://github.com/OCA/OpenUpgrade/issues?q=is%3Aopen++is%3Aissue+%22Migration+to+version%22+"
